modules/transactionpool/update.go

```diff
@@ -33,17 +32,87 @@ func (tp *TransactionPool) ProcessConsensusChange(cc modules.ConsensusChange) {
 				tp.log.Println("ERROR: could not delete a transaction:", err)
 			}
 		}
+
+		// Pull the transactions out of the fee summary. For estimating only
+		// over 10 blocks, it is extermely likely that there will be more
+		// applied blocks than reverted blocks, and if there aren't (a height
+		// decreasing reorg), there will be more than 10 applied blocks.
+		if len(tp.txnsPerBlock) > 0 {
+			tailOffset := uint64(0)
+			for i := 0; i < len(tp.txnsPerBlock)-1; i++ {
+				tailOffset += tp.txnsPerBlock[i]
+			}
+
+			// Strip out all of the transactions in this block.
+			tp.recentConfirmedFees = tp.recentConfirmedFees[:tailOffset]
```

Comment:
it's not clear to me how `recentConfirmedFees` and `tailOffset` are related. Probably want a sanity check to ensure that `tailOffset <= len(tp.recentConfirmedFees)`